types.go

```diff
@@ -282,10 +287,26 @@ func ByRevision[Obj any](rev uint64) Query[Obj] {
 
 // Index implements the indexing of objects (FromObjects) and querying of objects from the index (FromKey)
 type Index[Obj any, Key any] struct {
-	Name       string
+	// Name of the index
+	Name string
+
+	// FromObject extracts key(s) from the object. The key set
+	// can contain 0, 1 or more keys.
 	FromObject func(obj Obj) index.KeySet
-	FromKey    func(key Key) index.Key
-	Unique     bool
+
+	// FromKey converts the index key into a raw key.
+	// With this we can perform Query() against this index with
+	// the [Key] type.
+	FromKey func(key Key) index.Key
+
+	// FromString is an optional conversion from string to a raw key.
+	// If implemented allows script commands to query with this index.
+	FromString func(key string) (index.Key, error)
```

Comment:
I wonder if we want to make this mandatory or not. Would be a breaking change if we require it.
Might also make sense to have constructors for `Index` instead of having to fill in the struct.
I'm inclined to not make this mandatory as of yet as I want these changes to be non-breaking for now.